### PR TITLE
V2 content is application/json (not form-data)

### PR DESCRIPTION
**Pull request by malachy-mcconnell**

### CODE REVIEW

File: src/Connector/APIWrapper.php [hunk 1]
```diff
@@ -473,14 +486,21 @@ public function validateCustomerAddress($address) {
   }
 
   /**
+   * Please note this is not a V2 endpoint.
```
Comment (miromichalicka) [inline src/Connector/APIWrapper.php]: this should be in inferface

**Comment by malachy-mcconnell** [comment]
152 Code-sniff fails. I will work on those and push a new commit before merging.

**Comment by miromichalicka** [comment]
@malachy-mcconnell did you try something like this? https://gist.github.com/DragonBe/24761f350984c35b73966809dd439135 Currently all tests are skipped because our travis doesn't support that extension. Besides I'm not sure if we should add tests which are simply testing validity of json.

**Comment by malachy-mcconnell** [comment]
Hi Miro, thank you very much for fixing the CS. That was a huge help to me.

I did study the DragonBe suggested pattern. I think it works by declaring an interface for the final class. We could write an interface for the `final class client` but creating an interface grants an invitation to anyone to write a client class from the interface, circumventing the `final` stipulation.

In fact I would like to remove the `final` from the client class. I think that is the best solution. What is the point of this class being final? (other than indicating to developers 'we would prefer it if you did not extend this class' and wrecking our unit test)

Alternatively we could add the PECL module into our Travis build.

At the end of the day, maybe we say these unit tests are for developer desktop testing only and are skipped at CI.

**Pull request merged**